### PR TITLE
cmd/bootnode: enable prometheus

### PR DESCRIPTION
**Pull request by corverroos**
Enables prometheus metrics in the bootnode.

category: feature
ticket: #1506

### CODE REVIEW

File: cmd/bootnode/bootnode.go [hunk 1]
```diff
@@ -86,113 +84,80 @@ func Run(ctx context.Context, config Config) error {
...
+	promRegistry, err := promauto.NewRegistry(labels)
+	if err != nil {
+		return err
+	}
 
-		tcpNode, err := p2p.NewTCPNode(ctx, config.P2PConfig, key, p2p.NewOpenGater(), libp2p.ResourceManager(mgr))
-		if err != nil {
-			p2pErr <- errors.Wrap(err, "new tcp node")
+	// Start serving HTTP: ENR and monitoring.
+	serverErr := make(chan error, 2)
```
Comment (dB2510) [inline cmd/bootnode/bootnode.go:102]: why "2"?

File: cmd/bootnode/bootnode.go [hunk 2]
```diff
@@ -86,113 +84,80 @@ func Run(ctx context.Context, config Config) error {
...
 	log.Info(ctx, "Discv5 UDP bootnode started",
 		z.Str("p2p_udp_addr", config.P2PConfig.UDPAddr),
 		z.Str("enr", localEnode.Node().String()),
 	)
-	log.Info(ctx, "Runtime ENR available via http",
-		z.Str("url", fmt.Sprintf("http://%s/enr", config.HTTPAddr)),
-	)
+	if config.HTTPAddr != "" {
+		log.Info(ctx, "Runtime ENR available via http",
+			z.Str("url", fmt.Sprintf("http://%s/enr", config.HTTPAddr)),
```
Comment (dB2510) [inline cmd/bootnode/bootnode.go:148]: config.HTTPAddr should be something like this `http://abc.xyz`, this should be 
`fmt.Sprintf("%s/enr", config.HTTPAddr)` i think

File: cmd/bootnode/p2p.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		if err := relaylog.SetLogLevel("relay", config.RelayLogLevel); err != nil {
+			return nil, errors.Wrap(err, "set relay log level")
+		}
+	}
+
+	// Increase resource limits
+	limiter := rcmgr.DefaultLimits
+	limiter.SystemBaseLimit.ConnsInbound = config.MaxConns
+	limiter.SystemBaseLimit.FD = config.MaxConns
+	limiter.TransientBaseLimit = limiter.SystemBaseLimit
```
Comment (dB2510) [inline cmd/bootnode/p2p.go:44]: nit: add a comment for each of these limits

**Review by dB2510 — APPROVED** [review]

File: cmd/bootnode.go [hunk 4]
```diff
@@ -54,10 +54,10 @@ func newBootnodeCmd(runFunc func(context.Context, bootnode.Config) error) *cobra
 }
 
 func bindBootnodeFlag(flags *pflag.FlagSet, config *bootnode.Config) {
-	flags.StringVar(&config.HTTPAddr, "bootnode-http-address", "127.0.0.1:3640", "Listening address (ip and port) for the bootnode http server serving runtime ENR")
-	flags.BoolVar(&config.AutoP2PKey, "auto-p2pkey", true, "Automatically create a p2pkey (ecdsa private key used for p2p authentication and ENR) if none found in data directory")
-	flags.BoolVar(&config.P2PRelay, "p2p-relay", true, "Enable libp2p tcp host providing circuit relay to charon clusters")
-	flags.StringVar(&config.RelayLogLevel, "p2p-relay-loglevel", "", "Libp2p circuit relay log level. E.g., debug, info, warn, error")
+	flags.StringVar(&config.HTTPAddr, "bootnode-http-address", "127.0.0.1:3640", "Listening address (ip and port) for the bootnode http server serving runtime ENR.")
+	flags.StringVar(&config.MonitoringAddr, "bootnode-monitoring-address", "127.0.0.1:3620", "Listening address (ip and port) for the prometheus and pprof monitoring http server.")
```
Comment (xenowits) [inline cmd/bootnode.go:58]:
```suggestion
	flags.StringVar(&config.MonitoringAddr, "bootnode-monitoring-address", "127.0.0.1:3620", "Listening address (ip and port) for the prometheus and pprof monitoring http server for bootnode")
```

**Review by xenowits — APPROVED** [review]

File: cmd/bootnode/bootnode.go (hunk 1 above)
Comment (corverroos) [reply cmd/bootnode/bootnode.go:102]: 2 servers

File: cmd/bootnode/bootnode.go (hunk 2 above)
Comment (corverroos) [reply cmd/bootnode/bootnode.go:148]: it shouldn't be:
```
	flags.StringVar(&config.HTTPAddr, "bootnode-http-address", "127.0.0.1:3640", "Listening address (ip and port) for the bootnode http server serving runtime ENR.")

```

Comment (corverroos) [reply cmd/bootnode/bootnode.go:148]: can add a ticket to do validation of all listen address flags in the repo if you want.

File: cmd/bootnode/p2p.go (hunk 3 above)
Comment (corverroos) [reply cmd/bootnode/p2p.go:44]: I just moved this. You are are welcome to add the comments if you want.

File: cmd/bootnode.go (hunk 4 above)
Comment (corverroos) [reply cmd/bootnode.go:58]: bit redundant?

**Pull request merged**